lib/index.js

```diff
@@ -42,63 +40,69 @@ function R(...params) {
 
   const usedSettings = Object.assign({}, defaultSettings, settings);
 
-  return (event, context, callback) => {
+  return async (event, context) => {
     const startTime = process.hrtime();
-    const data = event && event.body
-      ? JSON.parse(event.body)
-      : event.queryStringParameters || {};
+    const data = event && event.body ? JSON.parse(event.body) : event.queryStringParameters || {};
+    const transformed = await usedSettings.transform(data);
```

Comment:
There will be unhandled error thrown if there is an error in user-passed transform function